### PR TITLE
New keyword services

### PR DESCRIPTION
**Pull request by hbaptiste**
New keyword services

### CODE REVIEW

File: Config/route.yml [hunk 1]
```diff
@@ -688,6 +688,7 @@ bb.keyword.get_collection:
   requirements:
     _method: GET
 
+
```
Comment (mickaelandrieu) [inline Config/route.yml]: unused line, please remove

File: Rest/Controller/KeywordController.php [hunk 2]
```diff
@@ -52,7 +58,6 @@ public function getCollectionAction(Request $request, $start, $count, KeyWord $p
         if (null !== $term) {
             $results = $this->getKeywordRepository()->getLikeKeyWords($term);
         } else {
-            $parent = (null !== $parent) ? $parent : $this->getKeywordRepository()->getRoot();
```
Comment (mickaelandrieu) [inline Rest/Controller/KeywordController.php]: you don't need to update $parent value, then ?

File: Rest/Controller/KeywordController.php [hunk 3]
```diff
@@ -75,7 +80,59 @@ public function getCollectionAction(Request $request, $start, $count, KeyWord $p
...
+            throw new BadRequestHttpException('A keyword show be provided.');
+        }
+
+       /* delete only if keyword is not used */
+       if (!$keyword->getContent()->isEmpty()) {
+           throw new BadRequestHttpException(sprintf('Keyword `%s` is linked to a content', $keyword->getKeyWord()));
+       }
+        $this->getKeywordRepository()->delete($keyword);
+
+        return new Response('', 204);
```
Comment (mickaelandrieu) [inline Rest/Controller/KeywordController.php]: :+1:

File: Rest/Controller/KeywordController.php (hunk 2 above)
Comment (hbaptiste) [inline Rest/Controller/KeywordController.php]: what do you mean?

File: Rest/Controller/KeywordController.php [hunk 4]
```diff
@@ -88,6 +145,158 @@ private function addRangeToContent(Response $response, $collection, $start)
         return $response;
     }
 
+    private function patchSiblingAndParentOperation(KeyWord $keyword, &$operations)
+    {
+        $sibling_operation = null;
```
Comment (mickaelandrieu) [inline Rest/Controller/KeywordController.php]: I know it's boring, plus please use CamelCase instead:
`$siblingOperation` and `$parentOperation`

File: Rest/Controller/KeywordController.php [hunk 5]
```diff
@@ -88,6 +145,158 @@ private function addRangeToContent(Response $response, $collection, $start)
...
+                }
+            } catch (InvalidArgumentException $e) {
+                throw new BadRequestHttpException(sprintf('Invalid node move action: %s', $e->getMessage()));
+            }
+        }
+    }
+
+    /**
+     * Create a keyword object
+     * and if a parent is provided added has its last child.
```
Comment (mickaelandrieu) [inline Rest/Controller/KeywordController.php]: and if a parent is provided add it as its last child

File: Rest/Tests/Controller/KeywordControllerTest.php [hunk 6]
```diff
@@ -41,43 +41,43 @@ protected function setUp() {
...
+      $keywordCollection = json_decode($response->getContent(), true);
       $this->assertInternalType('array', $keywordCollection);
-      $this->assertCount(2, $keywordCollection); 
-      
-      $url = '/rest/1/keyword';
+      $this->assertCount(2, $keywordCollection);
+
+      $url = '/rest/2/keyword';
       $response = $this->sendRequest(self::requestGet($url, ["term" => "toto"]));
       $this->assertTrue($response->isOk(), sprintf('HTTP 200 expected, HTTP %s returned.', $response->getStatusCode()));
```
Comment (mickaelandrieu) [inline Rest/Tests/Controller/KeywordControllerTest.php]: thanks for tests !

**Comment by mickaelandrieu** [comment]
please some aesthetic conventions then it's ready for merge.

Nice job @hbaptiste !

**Comment by mickaelandrieu** [comment]
merged. ping @clementbonfils this need to be deployed now.

**Pull request merged**